src/test/scala/com/microsoft/hyperspace/index/E2EHyperspaceRulesTests.scala

```diff
@@ -684,6 +684,66 @@ class E2EHyperspaceRulesTests extends QueryTest with HyperspaceSuite {
     }
   }
 
+  test("Hybrid scan works well with modified data when globbing pattern is used.") {
+    withTempPathAsString { testPath =>
+      val absoluteTestPath = PathUtils.makeAbsolute(testPath)
+      val globPath = absoluteTestPath + "/*"
+      val p1 = absoluteTestPath + "/1"
+      import spark.implicits._
+      SampleData.testData
+        .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+        .limit(10)
+        .write
+        .parquet(p1)
+
+      // Create index with globbing pattern.
+      val df = spark.read.option(GLOBBING_PATTERN_KEY, globPath).parquet(globPath)
+      val indexConfig = IndexConfig("index", Seq("clicks"), Seq("query"))
+      hyperspace.createIndex(df, indexConfig)
+
+      // Append data to a new directory which matches the globbing pattern.
+      val p2 = absoluteTestPath + "/2"
+      SampleData.testData
+        .toDF("Date", "RGUID", "Query", "imprs", "clicks")
+        .limit(3)
+        .write
+        .parquet(p2)
+
+      val df2 = spark.read.parquet(globPath)
+      def filterQuery: DataFrame = df2.filter(df2("clicks") <= 2000).select(df2("query"))
+      val baseQuery = filterQuery
+      val basePlan = baseQuery.queryExecution.optimizedPlan
+      spark.enableHyperspace()
+
+      withSQLConf(IndexConstants.INDEX_HYBRID_SCAN_ENABLED -> "false") {
+        val filter = filterQuery
+        assert(basePlan.equals(filter.queryExecution.optimizedPlan))
+      }
+
+      withSQLConf(IndexConstants.INDEX_HYBRID_SCAN_ENABLED -> "true") {
+        val filter = filterQuery
+        val planWithHybridScan = filter.queryExecution.optimizedPlan
+        assert(!basePlan.equals(planWithHybridScan))
+
+        // Check appended file is added to relation node or not.
+        val nodes = planWithHybridScan.collect {
+          case p @ LogicalRelation(fsRelation: HadoopFsRelation, _, _, _) =>
+            // Verify appended file is included or not.
+            assert(
+              fsRelation.location.inputFiles
+                .count(_.contains(absoluteTestPath.toUri.toString)) === 3)
+            // Verify number of index data files.
+            assert(fsRelation.location.inputFiles.count(_.contains("index")) === 4)
+            assert(fsRelation.location.inputFiles.length === 7)
```

Comment:
How should I understand 3, 4, and 7? Don't we need to check if `p2` exists, etc?

Reply:
thanks @imback82 , I cleaned up this test to check for p2, not-p1 and index files